chain/indexer/src/streamer/mod.rs

```diff
@@ -153,7 +154,33 @@ async fn build_streamer_message(
             receipt_execution_outcomes
                 .push(IndexerExecutionOutcomeWithReceipt { execution_outcome, receipt: receipt });
         }
-        indexer_shards[shard_id].receipt_execution_outcomes = receipt_execution_outcomes;
+        indexer_shards[shard_id].receipt_execution_outcomes = receipt_execution_outcomes.clone();
+
+        // Block #47317863 (ErdT2vLmiMjkRoSUfgowFYXvhGaLJZUWrgimHRkousrK) is the first block of upgraded
+        // protocol version on mainnet. In this block ExecutionOutcomes for restored Receipts appear.
+        // However the Receipts are not included in any Chunk. Indexer Framework needs to include them,
+        // so it was decided to artificially include the Receipts into the Chunk of the Block where
+        // ExecutionOutcomes appear.
+        // ref: https://github.com/near/nearcore/pull/4248
+        if &block.header.hash.to_string() == "ErdT2vLmiMjkRoSUfgowFYXvhGaLJZUWrgimHRkousrK" {
+            let protocol_config =
+                fetchers::fetch_protocol_config(&client, block.header.hash).await?;
+
+            if &protocol_config.chain_id == "mainnet" {
```

Comment:
Do we need this check? The hash we use is cryptographic

Reply:
We had hash collisions previously, so it's some sort of insurance from the indexer side. Do you insist on dropping this check?

Reply:
I am not so concerned about the hash collisions (if that happens, we are really doomed), but I still prefer to explicitly communicate the fact that we care about mainnet state here, so in the future we may consider just dropping it.